api/utils/keys/yubikey_test.go

```diff
@@ -53,6 +60,48 @@ func TestGetOrGenerateYubiKeyPrivateKey(t *testing.T) {
 	require.Equal(t, priv, retrieveKey)
 }
 
+func TestOverwriteSlot(t *testing.T) {
+	// This test expects a yubiKey to be connected with default PIV
+	// settings and will overwrite any PIV data on the yubiKey.
+	if os.Getenv("TELEPORT_TEST_YUBIKEY_PIV") == "" {
+		t.Skipf("Skipping TestGenerateYubiKeyPrivateKey because TELEPORT_TEST_YUBIKEY_PIV is not set")
+	}
+
+	ctx := context.Background()
+	resetYubikey(ctx, t)
+
+	oldStdin := prompt.Stdin()
+	t.Cleanup(func() { prompt.SetStdin(oldStdin) })
+
+	// Set a non-teleport certificate in the slot.
+	y, err := findYubiKey(0)
+	require.NoError(t, err)
+	yk, err := y.open()
+	require.NoError(t, err)
+
+	priv, err := rsa.GenerateKey(rand.Reader, 512)
+	require.NoError(t, err)
+	cert, err := metadataCertificateTemplate()
+	require.NoError(t, err)
+	cert.Subject.Organization = []string{"not-teleport"}
+	cert.PublicKey = priv.Public()
+	cert.Raw, err = x509.CreateCertificate(rand.Reader, cert, cert, priv.Public(), priv)
+	require.NoError(t, err)
```

Comment:
Could we do this using production code? Or, even better, test using public APIs?

Reply:
Can you clarify what you mean by public APIs?

Reply:
Using public/exported methods.

Reply:
9c8f581aea128cdc54fcc6018d10a9c4583d4a1d